pkg/api/types.go

```diff
@@ -1801,6 +1801,9 @@ type PodSpec struct {
 	// ServiceAccountName is the name of the ServiceAccount to use to run this pod
 	// The pod will be allowed to use secrets referenced by the ServiceAccount
 	ServiceAccountName string
+	// AutomountServiceAccountToken indicates whether a service account token should be automatically mounted.
+	// +optional
+	AutomountServiceAccountToken *bool
```

Comment:
This LGTM from an API perspective, @erictune do you concur based on the reasoning (we need to be able to opt out of a behavior we are already defaulting, so we can't use a new volume type to do this).